src/compiler/crystal/codegen/target_machine.cr

```diff
@@ -19,7 +22,7 @@ module Crystal
           features += "+vfp2"
         end
       else
-        raise "Unsupported arch for target triple: #{target_triple}"
+        raise TargetMachine::Error.new("Unsupported architecture for target triple: #{target_triple} (only x86, x86_64, and ARM are supported)")
```

Comment:
@ysbaddaden Just a note: In the case that the specific LLVM build does not include some otherwise supported target, it would not result in an error in this method.